tests/CMakeLists.txt

```diff
@@ -50,3 +50,43 @@ target_link_libraries(tests ${APPD_SDK_LINK_LIBS} ${CGREEN_LINK_LIBS})
 add_custom_target(run-tests COMMAND ./tests)
 
 add_dependencies(run-tests tests appdynamicsiotsdk)
+
+if (BUILD_COVERAGE)
+file (GLOB_RECURSE ALL_GCNO_FILES *.gcno)
+file (GLOB_RECURSE ALL_GCDA_FILES *.gcda)
+file (GLOB_RECURSE ALL_GCOV_FILES *.gcov)
+file (GLOB_RECURSE ALL_INFO_FILES *.info)
+
+foreach (gcnofile ${ALL_GCNO_FILES})
+    add_custom_command(
+    TARGET run-tests POST_BUILD
+    COMMAND gcov -s ${CMAKE_SOURCE_DIR}/sdk/src -o ${CMAKE_BINARY_DIR}/sdk/CMakeFiles/appdynamicsiotsdk.dir/src/ ${gcnofile}
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    VERBATIM)
+endforeach()
+
+add_custom_command(
+    TARGET run-tests POST_BUILD
+    COMMAND lcov --capture --directory . --output-file appd-iot-cpp-coverage-tmp.info
```

Comment:
lcov 1.13 version is needed. I had lcov 1.12 and it was looking for .da file instead of .gcda files.
lcov is not present by default and I installed it from http://ltp.sourceforge.net/coverage/lcov.php followed by make install.